### ISSUE TITLE
Make 'Allowed Values' for Properties translatable

### ISSUE THREAD

**Issue by lfcastel** [buildingSMART/bSDD#88]
Would it be possible to have property values that are translatable? Very convenient for multi-language countries.
![image](https://github.com/buildingSMART/bSDD/assets/73949503/48b963b0-6f51-45f3-9658-f630a7df8113)

**Comment by atomczak** [comment]
You mean the description of the value? The description can be translated, and the value needs to be the same in the model, otherwise it's no longer the same data.

**Comment by lfcastel** [comment]
> You mean the description of the value? The description can be translated, and the value needs to be the same in the model, otherwise it's no longer the same data.

I mean the value here. The thing is we have a code (acronym) which is actually what the user has to put. Then there's the full name and description. We use the code because in a multilingual country it is the easiest. But both name and description in our case wouldbe used in applications in different languages, but not as values

**Comment by atomczak** [comment]
So user, regardless of language, puts the same value in the model, right? Why can't you just translate the description? Can you give an example?

**Comment by araccaine** [comment]
Maybe I misunderstand this issue completely, but isn't `value` already stated as `Translatable` in the [JSON documentation](https://github.com/buildingSMART/bSDD/blob/3b19d4ac33c537f6fd51642a56d29e01a85370c5/Documentation/bSDD%20JSON%20import%20model.md#allowedvalue)?

> Code is a unique identification of the value (max 20 characters). It is required and, in most cases is the same as the value. It is needed to enable translations of Values or their Descriptions.

![grafik](https://github.com/buildingSMART/bSDD/assets/75312500/1efec977-2914-4775-b76e-c19c7141c4b2)

Using the `code` as identifier allows then to translate `value`. @lfcastel's example also totally makes sense to me: The *Form of the document* can be named differently in different languages, and the `code` allows us to pick the right value regardless of language. In the example above, *Bill of Quantities* could be *Mengenliste* in German. This translation is not obvious: A user would have to search for *Bill...* to get *Mengenliste*. This would make navigating the bSDD really inconvenient.

---

Not sure if related, but in this [presentation (page 7)](https://app.box.com/s/yhnjiq95676ft2a7c72cn3kikv3o28og), `name` is shown as translatable whereas `code` is used for identification. In the JSON documentation, the example is a little bit contradictory:

![grafik](https://github.com/buildingSMART/bSDD/assets/75312500/073e6846-a714-43c5-97d0-de7316e077bf)

Here `name` is the "Ifc-name" *IsExternal* and `code` is some arbitrary code *ifc-99088-01*. In the presentation, `code` represents the uniqe "IFC-name" and `name` is the "human-friendly name".

**Comment by lfcastel** [comment]
> So user, regardless of language, puts the same value in the model, right? Why can't you just translate the description? Can you give an example?

Yes because in a country with 2 official languages and people of both native languages working on the same project, we need a neutral value which is on our case would be English, more precise the English acronym. So if somebody uploads a bill of quantities on a CDE, our national convention says he/she should use BQ as one of the metadatafields. Depending on the user language this should the description 'bill of quantities' should then be displayed in Dutch or French.

I'd say it's comparable to having a ISO standard that has a code 19650-2, a name 'Information management using building information modelling - part 2: delivery phase of the assets' and a description.. both the name and the description should be translatable, but not the code.

as @araccaine mentions it is apparently translatable, but I always check the documentation so i wonder why i commented this, maybe the docs have been updated or the docs are not alligned with the import engine...

I think that i'm using the code as a value here, maybe it's not the right way to do it. So i say that Form can have different values (see the list) and every value has a code, value and description. But i don't see any alternative..

**Comment by atomczak** [comment]
Since property values are translatable, I'm closing the issue.
